apps/alb_monitor/lib/alb_monitor/monitor.ex

```diff
@@ -69,7 +69,7 @@ defmodule ALBMonitor.Monitor do
       "draining" ->
         log_info("shutdown")
         shutdown.()
-        {:stop, nil, state}
+        {:stop, :normal, state}
```

Comment:
question: any idea why returning `nil` is considered an error here? [The docs for `handle_info`](https://hexdocs.pm/elixir/GenServer.html#c:handle_info/2) say `reason :: any()` which would suggest it's legitimate. If it's not and the callback type signature is too permissive, should we consider writing a custom typespec for this function so Dialyzer can ensure we don't return anything but `:normal`?

Reply:
https://hexdocs.pm/elixir/Supervisor.html#module-exit-reasons-and-restarts has some more information:

>     `:normal` - in such cases, the exit won't be logged, there is no restart in transient mode, and linked processes do not exit
>    any other term - in such cases, the exit will be logged, there are restarts in transient mode, and linked processes exit with the same reason unless they're trapping exits

Reply:
Interesting, but this still doesn't say that returning `nil` causes an error. Seems a bit surprising to me.

Reply:
"nil" is in the "any other term" case.

Reply:
ah, so that explains why the process wasn't dead then! it was probably restarted.